browser/brave_ads/BUILD.gn

```diff
@@ -3,6 +3,132 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
+source_set("brave_ads") {
+  sources = [
+    "ads_service_delegate.h",
+    "ads_service_factory.h",
+    "analytics/p3a/brave_stats_helper.h",
+    "creatives/search_result_ad/creative_search_result_ad_tab_helper.h",
+    "tabs/ads_tab_helper.h",
+  ]
+
+  public_deps = [
+    "//base",
+    "//brave/components/brave_ads/browser",
+    "//chrome/browser/profiles:profile",
+    "//components/keyed_service/content",
+    "//components/prefs",
+    "//components/sessions:session_id",
+    "//content/public/browser",
+  ]
+
+  if (!is_android) {
+    public_deps += [ "//chrome/browser/ui:browser_list" ]
+  }
+}
+
+source_set("impl") {
+  visibility = [
+    ":*",
+    "//chrome/browser",
+  ]
+
+  sources = [
+    "ad_units/notification_ad/notification_ad_platform_bridge.h",
+    "ads_service_delegate.cc",
+    "ads_service_factory.cc",
+    "analytics/p3a/brave_stats_helper.cc",
+    "application_state/background_helper/background_helper.cc",
+    "application_state/background_helper/background_helper_holder.cc",
+    "application_state/background_helper/background_helper_holder.h",
+    "application_state/notification_helper/notification_helper.cc",
+    "application_state/notification_helper/notification_helper.h",
+    "application_state/notification_helper/notification_helper_impl.cc",
+    "application_state/notification_helper/notification_helper_impl.h",
+    "creatives/search_result_ad/creative_search_result_ad_tab_helper.cc",
+    "tabs/ads_tab_helper.cc",
+  ]
+
+  deps = [
+    ":brave_ads",
+    "//base",
+    "//brave/browser/brave_adaptive_captcha",
+    "//brave/browser/brave_ads/tooltips",
+    "//brave/browser/brave_rewards",
+    "//brave/browser/brave_rewards:util",
+    "//brave/browser/ui/brave_ads",
+    "//brave/components/brave_ads/browser",
+    "//brave/components/brave_ads/browser/application_state",
+    "//brave/components/brave_ads/content/browser",
+    "//brave/components/brave_ads/core/public:headers",
+    "//brave/components/brave_rewards/common",
+    "//brave/components/brave_search/common",
+    "//brave/components/p3a_utils",
+    "//chrome/browser:browser_process",
+    "//chrome/browser:browser_public_dependencies",
```

Comment:
This ensures that generated files from `//chrome/browser` run before this target. Any missing brave targets should go in `brave_chrome_browser_public_deps`